### DISCUSSION TITLE
What's the meaning of "5" in global install paths like, `pnpm/global/5/some-package`?

### DISCUSSION THREAD

**Discussion by AlanSl** [pnpm/pnpm#8202] [General]
With default settings, on PNPM 9, globally installed packages (on both my local Mac filesystem and an Alpine Linux Docker image) go into a directory named `5`, for example: `/Users/{me}/Library/pnpm/global/5/{package-name}` or `/usr/local/bin/global/5/{package-name}`.

What's the significance of this magic number `5`? Is it stable, does it depend on some other configuration somewhere?

If we've set `ENV PNPM_HOME=/usr/local/bin` in a Dockerfile, can we reliably expect to find globally installed files in `/usr/local/bin/global/5` or does it depend on something?

I've tried looking in the docs and existing issues and could find anything, but I found several other issues where users were quoting global install file paths that included `/global/5`.

**Comment by AlanSl** [comment]
Looking at this discussion https://github.com/orgs/pnpm/discussions/3351 it _looks_ like versions before pnpm@6 put their globally installed packages in a folder named for the major version number, then versions 6 and above just stuck with `5`? So the 5 signifies that it was installed by a version that >=5.x?

Is this stable for future versions and is this documented anywhere?

**Reply by zkochan** [reply]
it will change only in a new major version of pnpm if the new major version makes breaking changes to the structure of the directory
